ecommerce/serializers.py

```diff
@@ -479,6 +383,103 @@ def _update_coupons(cls, basket, product, coupons):
                 )
         return coupon_version
 
+    @classmethod
+    def _update_basket_data(
+        cls,
+        basket,
+        updated_product=None,
+        updated_program_run=None,
+        updated_run_ids=None,
+        coupon_version=None,
+        data_consents=None,
+        clear_all=False,
+    ):  # pylint: disable=too-many-arguments
+        """
+        Creates/updates/deletes Basket-related data
+
+        Args:
+            basket (models.Basket): The user's Basket
+            updated_product (models.Product): If provided, indicates a new Product that differs from the
+                previous Product associated with the Basket
+            updated_program_run (courses.models.ProgramRun): If provided, indicates a ProgramRun to associate
+                with the Product
+            updated_run_ids (iterable of int): If provided, indicates a new selection of CourseRun ids for
+                this basket
+            coupon_version (models.CouponVersion): A valid coupon to associate with this Basket. If None, all
+                coupon selections for the Basket are deleted
+            data_consents (iterable of int): Data consent ids
+            clear_all (boolean): If True, clears the basket, i.e.: all basket items, course run selections,
+                and coupon selections are deleted
+        """
+        with transaction.atomic():
```

Comment:
I'm wondering if this code would benefit from a `basket = Basket.objects.select_for_update().get(id=basket.id)` so that it locks on the basket for the duration of this transaction?

Reply:
The `Basket` object itself isn't being altered at all, just a bunch of related objects. Do you still see a benefit from locking on the `Basket` given that?

Reply:
It'd mean that concurrent requests wouldn't get you into a weird state, but if that's not a concern we can leave as-is.

Reply:
Good point. Just made that change